tests/test_smoke.py

```diff
@@ -682,6 +682,36 @@ def test_gcp_storage_mounts():
         run_one_test(test)
 
 
+@pytest.mark.cloudflare
```

Comment:
Nice! I also realized this might be a better way of selecting when to run R2 tests rather than using the `R2_AVAILABLE`. Is it possible to do the same in `TestStorageWithCredentials` below by replacing the parametrize  decorator like so:

```
    @pytest.mark.parametrize('store_type', [
        storage_lib.StoreType.S3, storage_lib.StoreType.GCS,
        pytest.param(storage_lib.StoreType.R2,
                     marks=pytest.mark.cloudflare)
    ])
```

(I'm not fully sure if the above code would work, you may need to dig around a bit)

The goal is to run cloudflare tests when `pytest tests/test_smoke.py --cloudflare` is run. You may need to modify conftest.py too.